petab/visualize/plotting_config.py

```diff
@@ -97,6 +97,18 @@ def plot_lowlevel(plot_spec: pd.Series,
 
         # construct errorbar-plots: noise specified above
         else:
+            # sort index for the case that indices of conditions and ms differ
+            # if indep_var='time', conditions is a numpy array,
+            # for indep_var=observable its a Series
+            if isinstance(conditions, np.ndarray):
+                conditions.sort()
+            elif isinstance(conditions, pd.core.series.Series):
+                conditions.sort_index(inplace=True)
+            else:
+                print('strange: conditions object is neither numpy nor '
```

Comment:
should this raise an error, i.e. the case never occur? then `raise ValueError("Strange: ...")`